# Nino 3 index

Down load the data HadISST_sst.nc from Canvas/Files/data. Load the data using the upward arrow icon on the top left. First, let's import the modules that we will use. Uploading the data take some time.

In [35]:
import matplotlib.pyplot as plt
import numpy as np
import xarray as xr

#### Exercise
Let's open the dataset and assign it as ds1. 

In [36]:
# Let's check the variable sst.
print(ds1.sst)

<xarray.DataArray 'sst' (time: 1812, latitude: 180, longitude: 360)>
[117417600 values with dtype=float32]
Coordinates:
  * time       (time) datetime64[ns] 1870-01-16T11:59:59.505615234 ... 2020-1...
  * latitude   (latitude) float32 89.5 88.5 87.5 86.5 ... -87.5 -88.5 -89.5
  * longitude  (longitude) float32 -179.5 -178.5 -177.5 ... 177.5 178.5 179.5
Attributes:
    standard_name:  sea_surface_temperature
    long_name:      sst
    units:          C
    cell_methods:   time: lat: lon: mean


In [37]:
# Let's check the time axis. The last time step is:
ds1.time.isel(time=-1)

<xarray.DataArray 'time' ()>
array('2020-12-16T12:00:00.000000000', dtype='datetime64[ns]')
Coordinates:
    time     datetime64[ns] 2020-12-16T12:00:00
Attributes:
    long_name:      Time
    standard_name:  time

#### Exercise
Select sst from 1980 to 2020 (use slicing) and define it as sst.

#### Exercise
Choose Nino3 region (5N-5S, 150W-90W)and calculate the average and save it as sst_nino3. This is Nino3 index. To choose and average, we can use </br>
sst.sel(latitude=slice(5,-5),longitude=slice(-150,-90)).mean(("latitude","longitude")) </br>
Latitude starts from 89.5, so slicing follows that order.

#### Exercises
We have to subtract the monthly mean values to calculate anomaly. First, let's calculate monthly average of sst_nino3 and save it as nino3_clim.We have to subtract the monthly mean values to calculate anomaly. First, let's calculate monthly average of sst_nino3 and save it as sst_clim.

#### Exercise
Save anomaly as nino3, which is sst_nino3(grouped by month) minus nino3_clim. 

Make a plot of nino3 (Nino 3 index).